File: Sems3/AD1/Assignment_1/q7.c
/* Find the smallest positive integer from a given array of positive and negative integer. */

#include <stdio.h>
#include <stdlib.h>

int main() {
    int arr[] = {9,5,8,2,5,-1,-8,-2,-5,9,-9};
    int array_length = sizeof(arr)/sizeof(arr[0]);
    int x=arr[0];
    for(int i=0;i<array_length;i++) {
        if(arr[i]<x && arr[i]>0) {
            x=arr[i];
        }
    }
    printf("%d\n",x);
}

File: Sems3/AD1/Assignment_1/q7.java
/* Find the smallest positive integer from a given array of positive and negative integer. */

import java.util.*;
public class q7 {
    public static void main(String[] args) {
        int[] a={4,1,2,3,-7,-5,0,8,5,};
        System.out.println(missingno(a));
    }

    static int missingno(int[] nums) {
        int i = 0;
        while (i < nums.length) {
            int correct = nums[i] - 1;
            if (nums[i] <= nums.length && nums[i] > 0 && nums[i] != nums[correct]) {
                int temp = nums[i];
                nums[i] = nums[correct];
                nums[correct] = temp;
            } else {
                i++;
            }
        }

        int count = 0;
        for (int j = 0; j < nums.length; j++) {
            if (nums[j] != j + 1) {
                return j + 1;
            }
        }
        return nums.length+1;
    }
}